src/pages/achievement/control/subcomponents/controlPanelTools/controlPanelUtils/PrerequisiteAdder.tsx

```diff
@@ -0,0 +1,61 @@
+import { Button } from '@blueprintjs/core';
+import React, { useState } from 'react';
+import AchievementInferencer from 'src/pages/achievement/dashboard/subcomponents/utils/AchievementInferencer';
+
+import { AchievementItem } from '../../../../../../commons/achievement/AchievementTypes';
+import AchievementSelector from './AchievementSelector';
+
+type PrerequisiteAdderProps = {
+  editableAchievement: AchievementItem;
+  setEditableAchievement: any;
+  inferencer: AchievementInferencer;
+  saveChanges: any;
+};
+
+function PrerequisiteAdder(props: PrerequisiteAdderProps) {
+  const { editableAchievement, setEditableAchievement, inferencer, saveChanges } = props;
+
+  const [isDialogOpen, setDialogOpen] = useState<boolean>(false);
+  const toggleDialogOpen = () => setDialogOpen(!isDialogOpen);
+
+  const nonPrerequisites = inferencer.listAvailablePrerequisites(editableAchievement.id);
+
+  const addPrerequisite = (prerequisiteID: number) => {
+    const newAchievement = editableAchievement;
+    newAchievement.prerequisiteIds.push(prerequisiteID);
+    return newAchievement;
+  };
+
+  const [addedPrerequisiteID, setAddedPrerequisiteID] = useState<number>(
+    nonPrerequisites.length === 0 ? 0 : nonPrerequisites[0]
+  );
```

Comment:
Small note: please add a comment that there is a default prerequisite; I expected it to initially have none.

Reply:
👍